nixos/modules/services/mail/dovecot.nix

```
@@ -631,30 +587,118 @@ in
     };
 
     services.dovecot2 = {
-      protocols =
-        optional cfg.enableImap "imap" ++ optional cfg.enablePop3 "pop3" ++ optional cfg.enableLmtp "lmtp";
+      settings =
+        {
+          base_dir = mkDefault baseDir;
+          sendmail_path = mkDefault "/run/wrappers/bin/sendmail";
+          mail_plugin_dir = mkDefault "/run/current-system/sw/lib/dovecot/modules";
+          mail_path = mkDefault "maildir:/var/spool/mail/%u";
+          mail_uid = mkIf (cfg.mailUser != null) (mkDefault cfg.mailUser);
+          mail_gid = mkIf (cfg.mailGroup != null) (mkDefault cfg.mailGroup);
```

```suggestion
          mail_uid = mkIf (cfg.mailUser != null) cfg.mailUser;
          mail_gid = mkIf (cfg.mailGroup != null) cfg.mailGroup;
```
They should not be desynced. In fact, I'd go further and allow nulls in cfg.settings and then have

```suggestion
          mail_uid = cfg.mailUser;
          mail_gid = cfg.mailGroup;
```